index.js

```diff
@@ -21,23 +28,52 @@ function hasYarn() {
 }
 
 function hasNpm() {
-  return !hasYarn();
+  if (cachedHasNPM !== undefined) return cachedHasNPM;
+
+  try {
+    const cmd = crossSpawn.sync('npm', ['--version']);
+    const version = cmd.stdout && cmd.stdout.toString().trim();
+    cachedHasNPM = !!version;
+  } catch (e) {
+    cachedHasNPM = false;
+  }
+
+  return cachedHasNPM;
 }
 
 function yarnOrNpm() {
-  return hasYarn() ? 'yarn' : 'npm'
+  if (cachedClient !== undefined) return cachedClient;
+
+  const pkgRoot = pkgDir.sync();
```

Comment:
i think it will be great to manually set `pkgRoot` via argument, if consumer already calculate package root. what do you think?

Reply:
This would need some changes to the CLI as well, so I'm going to punt to another PR when it arises as a problem.